assets/javascripts/discourse/components/query-result.js.es6

```diff
@@ -139,6 +140,61 @@ const QueryResultComponent = Ember.Component.extend({
     return transformedRelTable(groups);
   },
 
+  @computed(
+    "rows.[]",
+    "content.colrender.[]",
+    "content.result_count",
+    "colCount"
+  )
+  canShowGraph(rows, colrender, result_count, colCount) {
+    const hasTwoColumns = colCount === 2;
+    const secondColumnContainsNumber =
+      result_count > 0 && typeof rows[0][1] === "number";
+    const secondColumnContainsId = colrender[1] !== undefined;
+
+    return (
+      hasTwoColumns && secondColumnContainsNumber && !secondColumnContainsId
+    );
+  },
+
+  @computed("content.rows.[]", "content.colrender.[]")
+  graphLabels(rows, colRender) {
+    const labelSelectors = new Map([
+      ["user", (user) => user.username],
+      ["badge", (badge) => badge.name],
+      ["topic", (topic) => topic.title],
+      ["group", (group) => group.name],
+      ["category", (category) => category.name],
+    ]);
+
+    const relationName = colRender[0];
+
+    if (relationName) {
+      const lookupFunc = this[`lookup${relationName.capitalize()}`];
+      const labelSelector = labelSelectors.get(relationName);
+
+      if (lookupFunc && labelSelector) {
+        return rows.map((r) => {
+          const relation = lookupFunc.call(this, r[0]);
+          const label = labelSelector(relation);
+          return this._cutGraphLabel(label);
+        });
+      }
+    }
+
+    return rows.map((r) => this._cutGraphLabel(r[0]));
+  },
+
+  @computed("content.rows.[]")
+  graphValues(rows) {
+    return rows.map((r) => r[1]);
+  },
+
+  @computed("columnDispNames.[]")
```

Comment:
same minor nitpick here, but you can probably do:

```
graphDatasetName: reads("columnDispNames.1")
```

Reply:
This one is good too.